Sources/JSONRequest.swift

```diff
@@ -102,7 +105,7 @@ open class JSONRequest {
         updateRequest(payload: payload)
 
         let start = Date()
-        let session = networkSession()
+        let session = JSONRequest.urlSession ?? networkSession()
```

Comment:
```JSONRequest.urlSession``` should be set before a request is made i.e:
```
JSONRequest.urlSession = mySpecialURLSession()
JSONRequest.get(/* insert get request here */)
```
 I exposed it to the user to support the case that they wanted to inject their own session, but **we** really only use it for testing. I suppose it could be made internal much like the ```requireNetwork``` variable, but I would prefer to keep it as open just to give the user the option.